crates/bitcoind_rpc/tests/test_emitter.rs

```diff
@@ -189,15 +43,23 @@ fn block_to_chain_update(block: &bitcoin::Block, height: u32) -> local_chain::Up
 pub fn test_sync_local_chain() -> anyhow::Result<()> {
     let env = TestEnv::new()?;
     let mut local_chain = LocalChain::default();
-    let mut emitter = Emitter::from_height(&env.client, 0);
+    let tip = env.rpc_client().get_block_count()?;
+    let mut emitter = Emitter::from_height(env.rpc_client(), tip as u32);
 
     // mine some blocks and returned the actual block hashes
     let exp_hashes = {
-        let mut hashes = vec![env.client.get_block_hash(0)?]; // include genesis block
-        hashes.extend(env.mine_blocks(101, None)?);
+        let mut hashes = (0..=tip)
+            .map(|height| env.rpc_client().get_block_hash(height))
+            .collect::<Result<Vec<_>, _>>()?; // includes genesis block
+        hashes.extend(env.mine_blocks(101 - tip as usize, None, &env.bitcoind)?);
         hashes
     };
 
+    (0..tip).for_each(|height| {
+        let changeset = BTreeMap::from([(height as u32, Some(exp_hashes[height as usize]))]);
+        local_chain.apply_changeset(&changeset);
+    });
```

Comment:
This also needs a comment saying why.

Nit: I would iterate over `exp_hashes` (using `.enumerate` and `.take`). I would also use the `LocalChain::from_blocks` method to construct the `local_chain` here instead.